scripts/generate_psa_key_attributes.py

```diff
@@ -71,63 +122,235 @@ class PsaKeyLocation(IntEnum):
+
+        if self.location == PsaKeyLocation.LOCATION_CRACEN_KMU:
+            if self.identifier < 0 or self.identifier > 255:
+                raise ValueError("KMU slot number must be in range 0-255")
+
+            if self.cracen_usage == PsaCracenUsageScheme.NONE:
+                raise ValueError(
+                    "--cracen-usage must be set to other than NONE if location is LOCATION_CRACEN_KMU"
+                )
+
+        # AES and AEAD algorithms
+        if self.alg0 in (
+            PsaAlgorithm.CBC,
+            PsaAlgorithm.CBC_PKCS7,
+            PsaAlgorithm.CTR,
+            PsaAlgorithm.ECB,
+            PsaAlgorithm.CCM,
+            PsaAlgorithm.GCM,
+        ):
+            if self.key_type != PsaKeyType.AES:
+                raise ValueError(
+                    f"Algorithm {self.alg0.name} can only be used with the AES key type"
+                )
+            if self.key_bits not in (128, 192, 256):
+                raise ValueError(
+                    f"Algorithm {self.alg0.name} only supports 128-bit, 192-bit, and 256-bit keys"
+                )
+            if self.usage not in (
+                PsaKeyUsage.ENCRYPT,
+                PsaKeyUsage.DECRYPT,
+                PsaKeyUsage.ENCRYPT_DECRYPT,
+            ):
+                raise ValueError(
+                    f"Algorithm {self.alg0.name} can only be used with the ENCRYPT or DECRYPT key usage (or both)"
+                )
+
+        # Chacha20Poly1305 algorithm
+        elif self.alg0 == PsaAlgorithm.CHAHA20_POLY1305:
+            if self.key_type != PsaKeyType.CHACHA20:
+                raise ValueError(
+                    f"Algorithm {self.alg0.name} can only be used with the CHACHA20 key type"
+                )
+            if self.key_bits != 256:
+                raise ValueError(f"Algorithm {self.alg0.name} only supports 256-bit keys")
+            if self.usage not in (
+                PsaKeyUsage.ENCRYPT,
+                PsaKeyUsage.DECRYPT,
+                PsaKeyUsage.ENCRYPT_DECRYPT,
+            ):
+                raise ValueError(
+                    f"Algorithm {self.alg0.name} can only be used with the ENCRYPT or DECRYPT key usage (or both)"
+                )
+
+        # HMAC algorithm
+        elif self.alg0 == PsaAlgorithm.HMAC_SHA256:
+            if self.key_type != PsaKeyType.HMAC:
+                raise ValueError(
+                    f"Algorithm {self.alg0.name} can only be used with the HMAC key type"
+                )
+            if self.key_bits != 128:
+                raise ValueError(f"Algorithm {self.alg0.name} only supports 128-bit keys")
+            if self.usage not in (
+                PsaKeyUsage.SIGN,
+                PsaKeyUsage.SIGN_VERIFY,
+                PsaKeyUsage.VERIFY,
+            ):
+                raise ValueError(
+                    f"Algorithm {self.alg0.name} can only be used with the SIGN, SIGN_VERIFY, or VERIFY key usage"
```

Comment:
```suggestion
                    f"Algorithm {self.alg0.name} can only be used with the SIGN or VERIFY key usage (or both)"
```

Consistency with Encrypt/Decrypt (or both) elsewhere